In [13]:
import requests

In [14]:
url = "https://jsonplaceholder.typicode.com/posts/1"

#GET request
response = requests.get(url)
response

<Response [200]>

In [15]:
dir(response)

['__attrs__',
 '__bool__',
 '__class__',
 '__delattr__',
 '__dict__',
 '__dir__',
 '__doc__',
 '__enter__',
 '__eq__',
 '__exit__',
 '__firstlineno__',
 '__format__',
 '__ge__',
 '__getattribute__',
 '__getstate__',
 '__gt__',
 '__hash__',
 '__init__',
 '__init_subclass__',
 '__iter__',
 '__le__',
 '__lt__',
 '__module__',
 '__ne__',
 '__new__',
 '__nonzero__',
 '__reduce__',
 '__reduce_ex__',
 '__repr__',
 '__setattr__',
 '__setstate__',
 '__sizeof__',
 '__static_attributes__',
 '__str__',
 '__subclasshook__',
 '__weakref__',
 '_content',
 '_content_consumed',
 '_next',
 'apparent_encoding',
 'close',
 'connection',
 'content',
 'cookies',
 'elapsed',
 'encoding',
 'headers',
 'history',
 'is_permanent_redirect',
 'is_redirect',
 'iter_content',
 'iter_lines',
 'json',
 'links',
 'next',
 'ok',
 'raise_for_status',
 'raw',
 'reason',
 'request',
 'status_code',
 'text',
 'url']

In [16]:
response.status_code

200

In [17]:
if response.status_code == 200:
    data = response.json()
    print(data)
else:
    print("Failed to retrieve data")

{'userId': 1, 'id': 1, 'title': 'sunt aut facere repellat provident occaecati excepturi optio reprehenderit', 'body': 'quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto'}


In [18]:
url = "https://jsonplaceholder.typicode.com/posts"

# Data to send in the POST request

data = {
    "title": "New msg",
    "body" : "Welcome to the basics Of API",
    "userId" : 1
}

response = requests.post(url,json = data)
response.status_code

201

Common codes:

200 → OK (The request was successful, usually for GET)

201 → Created (The resource was successfully created — expected for POST requests)

400 → Bad Request (The server could not understand the request- invalid request )

401 → Unauthorized (Authentication is required)

404 → Not Found (URL or requested resource doesn’t exist)

500 → Server Error (Problem on the server side)

In [19]:
url = "https://jsonplaceholder.typicode.com/posts"

# Query parameters

params = {
    "id" : 7
}

# Append URL "https://jsonplaceholder.typicode.com/posts?id=7"

response = requests.get(url, params = params)
response

<Response [200]>

In [20]:
data = response.json()
print(data)

[{'userId': 1, 'id': 7, 'title': 'magnam facilis autem', 'body': 'dolore placeat quibusdam ea quo vitae\nmagni quis enim qui quis quo nemo aut saepe\nquidem repellat excepturi ut quia\nsunt ut sequi eos ea sed quas'}]


Coinbase API

In [26]:
app_id = "bb2cf49bfc3745ddb183c49355d62319"

url = f"https://openexchangerates.org/api/latest.json?app_id={app_id}"

response = requests.get(url)
response.status_code

200

In [27]:
if response.status_code == 200:
    data = response.json()
    print(data)
else:
    print("Failed to retrieve data")

{'disclaimer': 'Usage subject to terms: https://openexchangerates.org/terms', 'license': 'https://openexchangerates.org/license', 'timestamp': 1755262800, 'base': 'USD', 'rates': {'AED': 3.67293, 'AFN': 68.331908, 'ALL': 83.207863, 'AMD': 382.634934, 'ANG': 1.79, 'AOA': 911.955, 'ARS': 1298.5056, 'AUD': 1.534622, 'AWG': 1.8015, 'AZN': 1.7, 'BAM': 1.673054, 'BBD': 2, 'BDT': 121.454217, 'BGN': 1.673305, 'BHD': 0.377008, 'BIF': 2981.112129, 'BMD': 1, 'BND': 1.281694, 'BOB': 6.907525, 'BRL': 5.391, 'BSD': 1, 'BTC': 8.41903e-06, 'BTN': 87.426851, 'BWP': 13.378101, 'BYN': 3.323768, 'BZD': 2.00793, 'CAD': 1.37929, 'CDF': 2911.44961, 'CHF': 0.805409, 'CLF': 0.024611, 'CLP': 965.5, 'CNH': 7.183645, 'CNY': 7.1801, 'COP': 4050.26348, 'CRC': 505.848441, 'CUC': 1, 'CUP': 25.75, 'CVE': 94.324219, 'CZK': 20.932001, 'DJF': 178.01314, 'DKK': 6.382031, 'DOP': 61.531218, 'DZD': 129.67052, 'EGP': 48.3012, 'ERN': 15, 'ETB': 140.789381, 'EUR': 0.855061, 'FJD': 2.25435, 'FKP': 0.737378, 'GBP': 0.737378, 'GEL

In [34]:
url = f"https://openexchangerates.org/api/latest.json?app_id={app_id}"

# Query parameters

params = {
    "rates" : "AZN"
}

# Append URL "https://jsonplaceholder.typicode.com/posts?id=7"

response = requests.get(url, params = params)
response

<Response [200]>

In [35]:
data = response.json()
print(data["rates"]["AZN"])

1.7
